internal/resto/client.go

```diff
@@ -0,0 +1,117 @@
+package resto
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+const (
+	completeJobStatus string = "complete"
+	errorJobStatus    string = "error"
+)
+
+var jobStatuses = map[string]struct{}{
+	completeJobStatus: struct {}{},
+	errorJobStatus:    struct {}{},
+}
+
+var (
+	// ErrServerInaccessible represents error message when server is inaccessible
+	ErrServerInaccessible = errors.New("couldn't reach resto server")
+	// ErrNotFoundUser represents error message from server when user was not found
+	ErrNotFoundUser = errors.New("user was not found")
+)
+
+// Client http client
+type Client struct {
+	HTTPClient *http.Client
+	Host       string
+	Username   string
+	AccessKey  string
+}
+
+// New creates a new client
+func New(host, username, accessKey string, timeout int) Client {
+	return Client{
+		HTTPClient: &http.Client{Timeout: time.Duration(timeout) * time.Second},
+		Host:       host,
+		Username:   username,
+		AccessKey:  accessKey,
+	}
+}
+
+// GetJobDetails get job details
+func (c *Client) GetJobDetails(id string) (Details, error) {
+	request, err := createRequest(c.Host, c.Username, c.AccessKey, id)
+	if err != nil {
+		return Details{}, err
+	}
+
+	return makeRequest(c.HTTPClient, request)
+}
+
+// GetJobStatus gets job status
+func (c *Client) GetJobStatus(id string, pollDuration time.Duration) (Details, error) {
+	request, err := createRequest(c.Host, c.Username, c.AccessKey, id)
+	if err != nil {
+		return Details{}, err
+	}
+
+	ticker := time.NewTicker(pollDuration)
+	defer ticker.Stop()
+
+	var jobDetails Details
+
+	for range ticker.C {
+		jobDetails, err = makeRequest(c.HTTPClient, request)
+		if err != nil {
+			return Details{}, err
+		}
+
+		if _, ok := jobStatuses[jobDetails.Status]; ok {
+			jobDetails = jobDetails
+			break
+		}
+	}
+
+	return jobDetails, nil
+}
+
+func makeRequest(httpClient *http.Client, request *http.Request) (Details, error) {
+	response, err := httpClient.Do(request)
+	if err != nil {
+		return Details{}, err
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode >= http.StatusInternalServerError {
+		return Details{}, ErrServerInaccessible
+	}
+
+	if response.StatusCode == http.StatusNotFound {
+		return Details{}, ErrNotFoundUser
```

Comment:
Hm, this one's confusing. `404` could be many things (e.g. route is misconfigured), with "user not found" just being one of them. I'd suggest just returning it straight up as "job not found".

Reply:
API return 200 or 404 if we can't find the user. I changed this error to more common, that we couldn't find a job.